File: apps/android/app/src/main/java/site/smap/harubook/core/models/Quiz.kt
package site.smap.harubook.core.models

import kotlinx.serialization.Serializable

@Serializable
data class Quiz(
    val id: Int,
    val bookId: Int,
    val orderIndex: Int,
    val question: String,
    val choices: List<String>,
    val answerIndex: Int,
    val explanation: String? = null,
)

@Serializable
data class QuizzesResponse(
    val quizzes: List<Quiz>,
    val created: Boolean? = null,
)

File: apps/ios/HaruBook/Core/Models/Quiz.swift
import Foundation

/// 백엔드 `quizzes` 테이블 row 형식.
struct Quiz: Codable, Identifiable, Hashable, Sendable {
    let id: Int
    let bookId: Int
    let orderIndex: Int
    let question: String
    let choices: [String]
    let answerIndex: Int
    let explanation: String?
}

struct QuizzesResponse: Decodable {
    let quizzes: [Quiz]
    let created: Bool?
}
